activitysim/abm/models/tour_mode_choice.py

```diff
@@ -123,6 +135,61 @@ def tour_mode_choice_simulate(tours, persons_merged,
     primary_tours_merged['tour_purpose'] = \
         primary_tours_merged.tour_type.where(not_university, 'univ')
 
+    # if trip logsums are used, run trip mode choice
+    if model_settings.get('COMPUTE_TRIP_MODE_CHOICE_LOGSUMS', False):
```

Comment:
Is this functionality documented?

Reply:
In general, this code feels like it adds a lot of new functionality to this method. It would ease testing and understanding if this was introduced as a new standalone method called from this method.

Reply:
Good idea. Will work on this.

Reply:
Done. Broke it up into a few separate methods, incorporated under one larger one named `get_trip_mc_logsums_for_all_modes()`. Feedback welcomed.